### PR TITLE
Replace Location.local_path with path

### PR DESCRIPTION
**Pull request by Brendonovich**
see linked linear issue for deets

### CODE REVIEW

File: core/src/location/mod.rs [hunk 1]
```diff
@@ -268,10 +264,6 @@ pub async fn scan_location(
 	ctx: &LibraryContext,
 	location: indexer_job_location::Data,
 ) -> Result<(), LocationError> {
-	if location.local_path.is_none() {
-		return Err(LocationError::MissingLocalPath(location.id));
-	};
```
Comment (fogodev) [inline core/src/location/mod.rs:273]: Should we check that the location `node_id` is equal current node id?

File: core/prisma/schema.prisma [hunk 2]
```diff
@@ -95,8 +95,9 @@ model Location {
     id                     Int      @id @default(autoincrement())
     pub_id                 Bytes    @unique
     node_id                Int
-    name                   String?
-    local_path             String?
+    name                   String
+    path                   String
+    // local_path             String?
```
[the remark this answers is not in the document]
Comment (Brendonovich) [reply core/prisma/schema.prisma:100]: nah that's a mistake

File: core/src/location/error.rs [hunk 3]
```diff
@@ -25,8 +25,8 @@ pub enum LocationError {
 	NotDirectory(PathBuf),
 	#[error("Could not find directory in Location (path: {0:?})")]
 	DirectoryNotFound(String),
-	#[error("Missing local_path (id: {0})")]
-	MissingLocalPath(i32),
+	// #[error("Missing local_path (id: {0})")]
+	// MissingLocalPath(i32),
```
[the remark this answers is not in the document]
Comment (Brendonovich) [reply core/src/location/error.rs:29]: We may need an alternative error for if a location isn't on the current node, but this one can definitely go

File: core/src/location/manager/watcher/utils.rs [hunk 4]
```diff
@@ -51,85 +52,85 @@ pub(super) fn check_event(event: &Event, ignore_paths: &HashSet<PathBuf>) -> boo
...
 	event: Event,
-	library_ctx: LibraryContext,
+	library_ctx: &LibraryContext,
 ) -> Result<(), LocationManagerError> {
-	if let Some(ref location_local_path) = location.local_path {
-		inner_create_file(location.id, location_local_path, event, &library_ctx).await
-	} else {
-		Err(LocationManagerError::LocationMissingLocalPath(location.id))
+	if location.node_id != library_ctx.node_local_id {
+		return Err(LocationManagerError::LocationMissingLocalPath(location.id));
```
[the remark this answers is not in the document]
Comment (Brendonovich) [reply core/src/location/manager/watcher/utils.rs:111]: Yeah you're right

File: core/src/location/mod.rs (hunk 1 above)
Comment (Brendonovich) [reply core/src/location/mod.rs:273]: yeah we should

**Review by fogodev — APPROVED** [review]
LGTM

**Review by fogodev — APPROVED** [review]

**Pull request merged**